blackjax/kernels.py

```diff
@@ -463,3 +464,67 @@ def step_fn(rng_key: PRNGKey, state):
             )
 
         return SamplingAlgorithm(init_fn, step_fn)
+
+
+class imcmc:
```

Comment:
I think this is too general to be a high-level kernel. It is fine to only implement internals.